packages/front-end/components/SchemaBrowser/EditSqlModal.tsx

```diff
@@ -217,24 +230,40 @@ export default function EditSqlModal({
             <div className="bg-light p-1">
               <div className="row align-items-center">
                 <div className="col-auto">
-                  <Tooltip
-                    body="You do not have permission to run test queries"
-                    shouldDisplay={!canRunQueries}
-                  >
-                    <Button
-                      color="primary"
-                      className="btn-sm"
-                      onClick={handleTestQuery}
-                      loading={testingQuery}
-                      disabled={!canRunQueries}
-                      type="button"
+                  <Flex align="center">
+                    <Tooltip
+                      body="You do not have permission to run test queries"
+                      shouldDisplay={!canRunQueries}
                     >
-                      <span className="pr-2">
-                        <FaPlay />
-                      </span>
-                      Test Query
-                    </Button>
-                  </Tooltip>
+                      <Button
+                        color="primary"
+                        className="btn-sm"
+                        onClick={handleTestQuery}
+                        loading={testingQuery}
+                        disabled={!canRunQueries}
+                        type="button"
+                      >
+                        <span className="pr-2">
+                          <FaPlay />
+                        </span>
+                        Test Query
+                      </Button>
+                    </Tooltip>
+                    {canFormat ? (
```

Comment:
Now hiding the button if we know we can't correctly format the query based on the datasource type.